model/execution/DeliveryExecutionManagerService.php

```diff
@@ -238,4 +236,60 @@ public function setExtraTime($deliveryExecutions, $extraTime = 0, $extendedTime
         return $result;
     }
 
+    public function adjustTimers(array $deliveryExecutions, $seconds)
+    {
+        $result = ['processed' => [], 'unprocessed' => []];
+
+        /** @var TimerAdjustmentServiceInterface $timerAdjustmentService */
+        $timerAdjustmentService = $this->getServiceLocator()->get(TimerAdjustmentServiceInterface::SERVICE_ID);
+
+        /** @var DeliveryExecution $deliveryExecution */
+        foreach ($deliveryExecutions as $deliveryExecution) {
+            if (is_string($deliveryExecution)) {
+                $deliveryExecution = $this->getDeliveryExecutionById($deliveryExecution);
+            }
+
+            $success = false;
+            if ($this->isCorrectDeliveryExecutionStateForAdjustment($deliveryExecution)) {
+                if ($seconds > 0) {
+                    $success = $timerAdjustmentService->increase($deliveryExecution, $seconds);
+                } else {
+                    $success = $timerAdjustmentService->decrease($deliveryExecution, abs($seconds));
+                }
+            }
+
+            if ($success) {
+                $result['processed'][$deliveryExecution->getIdentifier()] = true;
+            } else {
+                $result['unprocessed'][$deliveryExecution->getIdentifier()] = false;
+            }
+        }
+
+        return $result;
+    }
+
+    /**
+     * @param DeliveryExecutionInterface $deliveryExecution
+     * @return bool
+     */
+    private function isCorrectDeliveryExecutionStateForAdjustment($deliveryExecution)
```

Comment:
Maybe we could rename it to something shorter like `canAdjustTimer` or `timerAdjustmentAllowed`. We may have some restrictions not related to state later.

Reply:
Done